besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java

```diff
@@ -2326,11 +2326,33 @@ public void miningParametersAreCaptured() throws Exception {
         .isEqualTo(BytesValue.fromHexString(extraDataString));
   }
 
+  @Test
+  public void pruningIsEnabledIfSyncModeIsFast() {
+    parseCommand("--sync-mode", "FAST");
+
+    verify(mockControllerBuilder).pruningEnabled(true);
+    verify(mockControllerBuilder).build();
+
+    assertThat(commandOutput.toString()).isEmpty();
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void pruningIsDisabledIfSyncModeIsFull() {
+    parseCommand("--sync-mode", "FULL");
+
+    verify(mockControllerBuilder).pruningEnabled(false);
+    verify(mockControllerBuilder).build();
+
+    assertThat(commandOutput.toString()).isEmpty();
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
   @Test
   public void pruningIsEnabledWhenSpecified() throws Exception {
```

Comment:
Probably worth having separate tests that explicitly enable and explicitly disable pruning.